### PR TITLE
Show errors when find unsupported framework in project.json

### PR DESCRIPTION
**Pull request by ChengTian**
parent https://github.com/aspnet/dnx/issues/1896

@davidfowl @muratg @troydai 

Error in VS:
capture1

Error in command-line tool (`dnu restore`):
capture

Error in command-line tool (`dnu build`):
capture2

### CODE REVIEW

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 1]
```diff
@@ -243,11 +243,18 @@ public RestoreCommand(FrameworkName fallbackFramework)
...
-            if (!Runtime.Project.TryGetProject(projectJsonPath, out project))
+            var diagnostics = new List<ICompilationMessage>();
+            if (!Runtime.Project.TryGetProject(projectJsonPath, out project, diagnostics))
             {
                 throw new Exception("TODO: project.json parse error");
             }
 
+            if (diagnostics.HasErrors())
+            {
+                var error = diagnostics.First(x => x.Severity == CompilationMessageSeverity.Error);
```
Comment (muratg) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Could there be multiple errors? What should be the output in that case?

File: src/Microsoft.Framework.Runtime/Project.cs [hunk 2]
```diff
@@ -312,7 +312,7 @@ internal static Project GetProjectFromStream(Stream stream, string projectName,
                         key => repository.ValueAsString(key).Value);
             }
 
-            project.BuildTargetFrameworksAndConfigurations(rawProject);
+            project.BuildTargetFrameworksAndConfigurations(rawProject, diagnostics);
```
Comment (muratg) [inline src/Microsoft.Framework.Runtime/Project.cs]: It doesn't look like we're doing anything with the `diagnostics` after this call. Should there be a check before we populate the dependencies below?

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs (hunk 1 above)
Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Good point! It's possible that we have more than one unknown framework. I should turn to `AggregateException` here, in order to represent multiple errors. Fixed in follow-up commit.

File: src/Microsoft.Framework.Runtime/Project.cs (hunk 2 above)
Comment (ChengTian) [inline src/Microsoft.Framework.Runtime/Project.cs]: This `diagnostics` is for output purpose. The check will be done by caller of this method.

Comment (muratg) [inline src/Microsoft.Framework.Runtime/Project.cs]: :+1:

File: src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs [hunk 3]
```diff
@@ -243,11 +243,20 @@ public RestoreCommand(FrameworkName fallbackFramework)
...
             {
                 throw new Exception("TODO: project.json parse error");
             }
 
+            if (diagnostics.HasErrors())
+            {
+                var exceptions = diagnostics
+                    .Where(x => x.Severity == CompilationMessageSeverity.Error)
+                    .Select(x => new Exception(x.FormattedMessage));
+                throw new AggregateException(exceptions);
```
Comment (davidfowl) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: Don't throw exceptions from command line tools. It looks bad :smile: . I'm also not sure if we should fail up front here. Maybe we can just warn about unknown frameworks during restore.

Comment (ChengTian) [inline src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs]: I cannot see any benefit postponing the failure to build/publish. so let's fail fast to eliminate unnecessary confusion. Just pushed the fix.

**Comment by ChengTian** [comment]
Fixed behavior of `dnu restore`, instead of throwing exceptions, we show something like:
capture

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**